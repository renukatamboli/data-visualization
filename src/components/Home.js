import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";
import DataTable from "./DataTable";
import Graphs from "./Graphs";
const styles = (theme) => ({
  graphs: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    padding: theme.spacing(1),
  },
});
class Home extends Component {
  state = {
    data: "",
  };
  getData = (e) => {
    var formData = new FormData();
    formData.append("file", e.target.files[0]);
    return fetch("http://localhost:8080/api/csv/upload", {
      method: "POST",
      body: formData,
    })
      .then((response) => response.json())
      .then((response) => {
        this.setState({ data: response });
      })
      .catch((error) => {
        console.log("Unable to fetch data");
      });
  };
  render() {
    const { classes } = this.props;
    return (
      <div data-testid="home">
        <br />
        {this.state.data !== "" ? (
          <>
            <Graphs data={this.state.data} />
            <DataTable data={this.state.data} />{" "}
          </>
        ) : (
          <input
            accept="image/*"
            id="fileUpload"
            multiple
            type="file"
            onChange={this.getData}
            data-testid="inputFile"
          />
        )}
      </div>
    );
  }
}
export default withStyles(styles)(Home);
