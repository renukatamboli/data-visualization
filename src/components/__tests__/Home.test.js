import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Home from "../Home";
let originalFetch;
let promise;

jest.mock("react-apexcharts", () => {
  return {
    __esModule: true,
    default: () => {
      return <div />;
    },
  };
});

beforeEach(() => {
  originalFetch = global.fetch;
  const promise = Promise.resolve({
    json: () =>
      Promise.resolve([
        {
          TotalValue: "358752.94",
          Category: "A",
          Year: "01 Jan 2014",
          Quantity: "68646.896",
        },
        {
          TotalValue: "2432036.26",
          Category: "B",
          Year: "02 Jan 2014",
          Quantity: "400296.928",
        },
        {
          TotalValue: "2375685.48",
          Category: "C",
          Year: "03 Jan 2014",
          Quantity: "398729.693",
        },
        {
          TotalValue: "35656.8",
          Category: "D",
          Year: "04 Jan 2014",
          Quantity: "13525.89",
        },
      ]),
  });
  global.fetch = jest.fn(() => promise);
});

afterEach(() => {
  global.fetch = originalFetch;
});

test("match snapshot", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});

test("fetch call", async () => {
  const file = new File(["test"], "test.csv", { type: "csv" });
  render(<Home />);
  const inputElem = screen.getByTestId("inputFile");
  userEvent.upload(inputElem, file);
  await waitFor(() => {
    expect(inputElem).not.toBeInTheDocument();
  });
  const graph = screen.getByTestId("graph");
  const dataTable = screen.getByTestId("table");
  expect(graph).toBeInTheDocument();
  expect(dataTable).toBeInTheDocument();
});
