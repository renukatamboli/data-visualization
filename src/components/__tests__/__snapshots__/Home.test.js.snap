// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div>
  <div
    data-testid="home"
  >
    <br />
    <input
      accept="image/*"
      data-testid="inputFile"
      id="contained-button-file"
      multiple=""
      type="file"
    />
  </div>
</div>
`;
